pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,18 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
 	return l.Status
 }
 
+// +kubebuilder:validation:Enum=name;shard
+type AdditionalLabelSelector string
+
+type TopologySpreadConstraint struct {
+	*v1.TopologySpreadConstraint `json:",inline"`
```

Comment:
hmm why a pointer?

Reply:
By mistake

Reply:
When I'm not using a pointer I'm getting those errors.

```# github.com/prometheus-operator/prometheus-operator/pkg/client/applyconfiguration/monitoring/v1
pkg/client/applyconfiguration/monitoring/v1/topologyspreadconstraint.go:42:14: cannot use &value (value of type *int32) as int32 value in assignment
pkg/client/applyconfiguration/monitoring/v1/topologyspreadconstraint.go:50:18: cannot use &value (value of type *string) as string value in assignment
pkg/client/applyconfiguration/monitoring/v1/topologyspreadconstraint.go:58:24: cannot use &value (value of type *"k8s.io/api/core/v1".UnsatisfiableConstraintAction) as "k8s.io/api/core/v1".UnsatisfiableConstraintAction value in assignment```

Reply:
Hmm I don't see the same thing :thinking:

Reply:
After remove the `*` mark and run make generate I got this issue.

Reply:
@simonpasquier any idea?

Reply:
Have you tried to delete the generated Go files first?

Reply:
I think that I managed to workaround the issue by defining a few intermidiate structs:

```go
type TopologySpreadConstraint struct {
       CoreV1TopologySpreadConstraint `json:",inline"`
 
        //+optional
        // Defines what Prometheus Operator managed labels should be added to labelSelector on the topologySpreadConstraint.
        AdditionalLabelSelectors *AdditionalLabelSelector `json:"additionalLabelSelectors,omitempty"`
 }

type CoreV1TopologySpreadConstraint v1.TopologySpreadConstraint
```